server/src/test/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/data/StarTreeFileFormatsTests.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.fileformats.data;
+
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.opensearch.index.compositeindex.datacube.startree.fileformats.StarTreeWriter;
+import org.opensearch.index.compositeindex.datacube.startree.fileformats.meta.StarTreeMetadata;
+import org.opensearch.index.compositeindex.datacube.startree.node.InMemoryTreeNode;
+import org.opensearch.index.compositeindex.datacube.startree.node.StarTree;
+import org.opensearch.index.compositeindex.datacube.startree.node.StarTreeNode;
+import org.opensearch.test.OpenSearchTestCase;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Queue;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class StarTreeFileFormatsTests extends OpenSearchTestCase {
+
+    private IndexOutput dataOut;
+    private IndexInput dataIn;
+    private Directory directory;
+
+    @Before
+    public void setup() throws IOException {
+        directory = newFSDirectory(createTempDir());
+    }
+
+    public void test_StarTreeNode() throws IOException {
+
+        dataOut = directory.createOutput("star-tree-data", IOContext.DEFAULT);
+        Map<Long, InMemoryTreeNode> levelOrderStarTreeNodeMap = new LinkedHashMap<>();
+        InMemoryTreeNode root = generateSampleTree(levelOrderStarTreeNodeMap);
+        long starTreeDataLength = StarTreeWriter.writeStarTree(dataOut, root, 7, "star-tree");
+
+        // asserting on the actual length of the star tree data file
+        assertEquals(starTreeDataLength, 247);
+        dataOut.close();
+
+        dataIn = directory.openInput("star-tree-data", IOContext.READONCE);
+
+        StarTreeMetadata starTreeMetadata = mock(StarTreeMetadata.class);
+        when(starTreeMetadata.getDataLength()).thenReturn(starTreeDataLength);
+        when(starTreeMetadata.getDataStartFilePointer()).thenReturn(0L);
+        StarTree starTree = new StarTree(dataIn, starTreeMetadata);
+
+        StarTreeNode starTreeNode = starTree.getRoot();
+        Queue<StarTreeNode> queue = new ArrayDeque<>();
+        queue.add(starTreeNode);
+
+        while ((starTreeNode = queue.poll()) != null) {
+
+            // verify the star node
```

Comment:
Can we add few tests emulating search behavior ? randomly select nodes between 0 and length of children node at each dimension as we traverse and try querying those nodes in both in memory and in file formats.

Assertions look good, except i just want some randomness instead of going through all nodes via BFS.